tracee-ebpf/main.go

```diff
@@ -955,13 +953,53 @@ func prepareEventsToTrace(eventFilter *tracee.StringFilter, setFilter *tracee.St
 	return res, nil
 }
 
-func checkRequiredCapabilities(caps capability.Capabilities) error {
-	if !caps.Get(capability.EFFECTIVE, capability.CAP_SYS_ADMIN) {
-		return fmt.Errorf("insufficient privileges to run: missing CAP_SYS_ADMIN")
+const bpfCapabilitiesMinKernelVersion = "5.8"
+
+// ensureCapabilities makes sure the program has just the required capabilities to run
+func ensureCapabilities(OSInfo *helpers.OSInfo) error {
```

Comment:
How do you feel about making this function more readable and easy to maintain with something like:

```Go
func ensureCapabilities(OSInfo *helpers.OSInfo) error {
        selfCap, err := getSelfCapabilities()
        if err != nil {
                return err
        }

        r := []capability.Cap {
            capability.CAP_IPC_LOCK,
            capability.CAP_SYS_PTRACE,
            capability.CAP_SYS_RESOURCE,
        }

        if OSInfo.CompareOSBaseKernelRelease("5.8") <= 0 {
                r = append(r, []capability.Cap{
                    capability.CAP_BPF,
                    capability.CAP_PERFMON,
                }...)
        } else {
                r = append(r, []capability.Cap{
                    capability.CAP_SYS_ADMIN,
                }...)
        }

        s := append(r, capability.CAP_SETPCAP)

        if err = checkRequiredCapabilities(selfCap, s); err != nil {
                return err
        }
        if err = dropUnrequiredCapabilities(selfCap, r); err != nil {
                return err
        }

        return nil
```

Long variable names won't always help reading the source code, despite telling what the variables are for (which can be followed by the function names being called).

Having each capability on its own line helps maintaining the code. Future changes/patches to this code can add/remove capabilities by simply adding/removing lines from it (instead of changing the entire line).

Reply:
About the lines breaks - I think you are more than right, this is much more readable.
About the names of the variables - I think that the roles of each of the startup and runtime capabilities are not so clear from the code itself, so calling them just 's' and 'r' is too short. However, the names I have given might be too long.
Is it ok with you that I will call them 'startup' and 'runtime'?

Reply:
yep, agreed.